### PR TITLE
Fix bug with incorrect toml configs merging

### PR DESCRIPTION
**Pull request by kgevorkyan**
### What's done:
* Fix bug in logic of `execFlags` merging, add test
* Move logic corresponding with toml processing into separate class

### CODE REVIEW

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/utils/TomlUtils.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+) =
+        try {
+            val pluginConfig = TomlDecoder.decode<T>(
+                serializer(),
+                fakeFileNode,
+                DecoderConf()
+            ).apply {
+                configLocation = this@createPluginConfig.toPath()
+            }
+            pluginConfig
```
Comment (petertrr) [inline save-core/src/commonMain/kotlin/org/cqfn/save/core/utils/TomlUtils.kt:89]:
```suggestion
            TomlDecoder.decode<T>(
                serializer(),
                fakeFileNode,
                DecoderConf()
            ).apply {
                configLocation = this@createPluginConfig.toPath()
            }
```

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt [hunk 2]
```diff
@@ -73,7 +62,11 @@ class Save(
                 reporter.beforeAll()
 
                 // discover plugins from the test configuration
-                discoverPluginsAndUpdateTestConfig(testConfig)
+                testConfig.pluginConfigs.clear()
```
Comment (petertrr) [inline save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt:65]: Did we not clear it before? Why is it needed now?

Comment (kgevorkyan) [reply save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt]: Yes, It should be empty. Removed

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/utils/TomlUtils.kt (hunk 1 above)
Comment (kgevorkyan) [reply save-core/src/commonMain/kotlin/org/cqfn/save/core/utils/TomlUtils.kt]: Thanks. Fixed

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt [hunk 3]
```diff
@@ -73,7 +62,11 @@ class Save(
...
+                testConfig.location.toString().createPluginConfigListFromToml().forEach {
```
[the remark this answers is not in the document]
Comment (kgevorkyan) [reply save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt]: Changed to `createPluginConfigListFromToml(testConfig.location)`

**Review by petertrr — APPROVED** [review]

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**